src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/UserAcspMembership.java

```diff
@@ -26,33 +29,67 @@
 @RestController
 public class UserAcspMembership implements UserAcspMembershipInterface {
 
-  private static final String PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN = "Please check the request and try again";
+  private static final String PLEASE_CHECK_THE_REQUEST_AND_TRY_AGAIN =
+      "Please check the request and try again";
   private static final Logger LOG =
-          LoggerFactory.getLogger(StaticPropertyUtil.APPLICATION_NAMESPACE);
+      LoggerFactory.getLogger(StaticPropertyUtil.APPLICATION_NAMESPACE);
   private final AcspMembersService acspMembersService;
   private final AcspMembershipService acspMembershipService;
+  private final UsersService usersService;
 
-  public UserAcspMembership(final AcspMembersService acspMembersService, final AcspMembershipService acspMembershipService) {
+  public UserAcspMembership(
+      final AcspMembersService acspMembersService,
+      final AcspMembershipService acspMembershipService,
+      final UsersService usersService) {
     this.acspMembersService = acspMembersService;
     this.acspMembershipService = acspMembershipService;
+    this.usersService = usersService;
   }
 
   @Override
   public ResponseEntity<ResponseBodyPost> addAcspMember(
-          @NotNull String xRequestId,
-          @NotNull String ericIdentity,
-          @Valid RequestBodyPost requestBodyPost
-  ) {
-    return null; // TODO(https://companieshouse.atlassian.net/browse/IDVA6-1149)
+      final String xRequestId,
+      final String requestingUserId,
+      final RequestBodyPost requestBodyPost) {
+    Optional<AcspMembersDao> requestingUserMembership =
+        acspMembersService.fetchActiveAcspMemberByUserIdAndAcspNumber(
+            requestingUserId, requestBodyPost.getAcspNumber());
+    if (requestingUserMembership.isEmpty()) {
```

Comment:
I don’t think we have a generic handler for forbidden i.e the others in this file: 
https://github.com/companieshouse/acsp-manage-users-api/blob/616962e889fb9be0193fe9650e03bfa217842fb7/src/main/java/uk/gov/companieshouse/acsp/manage/users/controller/ControllerAdvice.java#L27
Maybe it’s best to provide a message with the forbidden rather than throw an exception? Not too fussed - do we have a best practice or way we typically do this so we can be consistent?

Reply:
Now that you mention it, the only place we actually return Forbidden is in the interceptors. We generally return 400, 404, and 500 for all of the errors. For example, if you look at the requirements for your ticket, the AC's are to actually return 400 when the user does not have permission. I do think Forbidden is more appropriate, but I think this might just be the way Companies House does things. You'll probably have to change the 403s to throw a BadRequestRuntimeException or NotFoundRuntimeException, depending on what it says in the ticket.

Companies House has a standard Errors object for HTTP errors, and a pattern for mapping custom exception types (e.g. BadRequestRuntimeException) to Http responses via the ControllerAdvice. We generally want to stick to this pattern where possible.

Reply:
Hey - nice shout, have refactored all this in https://github.com/companieshouse/acsp-manage-users-api/pull/13/commits/acfe2d7438dd318ec3f28399538b90ff0b0d699c, thanks for the feedback.